metrics/web/test/common/presentation/navigation/metrics_page/metrics_page_factory_test.dart

```diff
@@ -2,167 +2,239 @@
 // that can be found in the LICENSE file.
 
 import 'package:metrics/auth/presentation/pages/login_page.dart';
-import 'package:metrics/common/presentation/navigation/constants/default_routes.dart';
 import 'package:metrics/common/presentation/navigation/metrics_page/metrics_page_factory.dart';
+import 'package:metrics/common/presentation/navigation/route_configuration/route_name.dart';
 import 'package:metrics/common/presentation/pages/loading_page.dart';
+import 'package:metrics/dashboard/presentation/models/dashboard_page_parameters_model.dart';
 import 'package:metrics/dashboard/presentation/pages/dashboard_page.dart';
 import 'package:metrics/debug_menu/presentation/pages/debug_menu_page.dart';
 import 'package:metrics/project_groups/presentation/pages/project_group_page.dart';
 import 'package:mockito/mockito.dart';
 import 'package:test/test.dart';
 
-import '../../../../test_utils/route_configuration_mock.dart';
 import '../../../../test_utils/route_name_mock.dart';
 
 void main() {
   group("MetricsPageFactory", () {
-    const path = 'path';
+    const unknownRouteName = 'unknown';
+    const pageParameters = DashboardPageParametersModel();
+    const dashboardRouteName = RouteName.dashboard;
+    const loadingRouteName = RouteName.loading;
+    const loginRouteName = RouteName.login;
+    const projectGroupsRouteName = RouteName.projectGroups;
+    const debugMenuRouteName = RouteName.debugMenu;
 
     final metricsPageFactory = MetricsPageFactory();
-    final routeConfiguration = RouteConfigurationMock();
     final routeName = RouteNameMock();
 
     tearDown(() {
-      reset(routeConfiguration);
       reset(routeName);
     });
 
     test(
-      ".create() returns the dashboard metrics page if the given route configuration is null",
+      ".create() returns the dashboard metrics page if the given route name is null",
       () {
-        final page = metricsPageFactory.create(null);
+        final page = metricsPageFactory.create(null, pageParameters);
 
         expect(page.child, isA<DashboardPage>());
       },
     );
 
     test(
-      ".create() returns the loading metrics page with a name equals to the given route configuration path",
+      ".create() returns the dashboard metrics page with the name equals to the dashbord route name value if the given route name is null",
       () {
-        const routeConfiguration = DefaultRoutes.loading;
-        final expectedName = routeConfiguration.path;
+        final page = metricsPageFactory.create(null, pageParameters);
 
-        final actualName = metricsPageFactory.create(routeConfiguration).name;
+        expect(page.name, dashboardRouteName.value);
+      },
+    );
 
-        expect(actualName, equals(expectedName));
+    test(
+      ".create() returns the dashboard metrics page with arguments equals to the given page parameters if the route name is null",
```

Comment:
🏷️ 
```suggestion
      ".create() returns the metrics page with arguments equals to the given page parameters if the given route name is null",
```
It seems like this method returns a `MetricsPage`, but not the `DashboardMetricsPage`. Please, apply this suggestion to all similar tests.